code/src/reducers/todos.js

```diff
@@ -0,0 +1,43 @@
+import { createSlice } from "@reduxjs/toolkit";
+import uniqid from "uniqid";
+
+const todos = createSlice({
+  name: "todos",
+  initialState: {
+    items: [],
+  },
+  reducers: {
+    addTodo: (store, action) => {
+      const newTodo = {
+        id: uniqid(),
+        text: action.payload,
+        isComplete: false,
+      };
+
+      store.items = [...store.items, newTodo];
+    },
+
+    toggleTodo: (store, action) => {
+      const updatedItems = store.items.map((item) => {
+        if (item.id === action.payload) {
+          const updatedTodo = {
+            ...item,
+            isComplete: !item.isComplete,
+          };
+          return updatedTodo;
+        } else {
+          return item;
+        }
+      });
+      store.items = updatedItems;
+    },
+    deleteTodo: (store, action) => {
+      const decreasedItems = store.items.filter(
+        (item) => item.id !== action.payload
+      );
+      store.items = decreasedItems;
+    },
+  },
+});
+
+export default todos;
```

Comment:
Hi Rebecca, I know you are in progress to finish this project but what a good projects so far!! Your code is easy to read and I love that you are using localstore and styledcomponents(i didnt)!!  The only thing you can look at is the length of the text in the text box. Even if you are still in progress to finish this project I see that you have already reced the requirements for blue level and also requirements from the other levels. Well Done! :D
- [ YES]  Am I able to understand the code easily?
- [NO ]  Does the code have lots of duplicates? Could it be tidied up?
- [YES, works]  Does it work or does anything look broken?
- [ YES]  Does it follow the general and blue requirements?

Reply:
Thank you for your nice comments, Jessica!